include/usb/usb_device.h

```diff
@@ -298,4 +298,56 @@ int usb_ep_read_wait(u8_t ep, u8_t *data, u32_t max_data_len,
  */
 int usb_ep_read_continue(u8_t ep);
 
+/**
+ * Callback function signature for transfer completion
+ */
+typedef void (*usb_transfer_callback)(u8_t ep, int status, size_t tsize);
+
+/* USB transfer flags */
+#define USB_TRANS_READ  1
+#define USB_TRANS_WRITE 2
+
+/**
+ * @brief Start a transfer
+ *
+ * Start a usb transfer to/from the data buffer.
+ *
+ * @param[in]  ep           Endpoint address corresponding to the one
+ *                          listed in the device configuration table
+ * @param[in]  data         Pointer to data buffer to write-to/read-from
+ * @param[in]  dlen         Size of data buffer
+ * @param[in]  flags        Transfer flags (USB_TRANS_READ, USB_TRANS_WRITE...)
+ * @param[in]  cb           Function called on transfer completion/failure.
+ *
+ * @return 0 on success, negative errno code on fail.
+ */
+int usb_transfer(u8_t ep, u8_t *data, size_t dlen, unsigned int flags,
+		 usb_transfer_callback cb);
+
+/**
+ * @brief Start a transfer and wait for completion
```

Comment:
Instead of "wait", please mention it as "block wait" which makes it a bit clear on the wait part.